src/cider/nrepl/middleware/info.clj

```diff
@@ -57,9 +59,44 @@
           blacklist
           util/transform-value))))
 
+(defn extract-class-from-compliment [ns-str context]
+  (when (and (seq ns-str)
+             (seq context))
+    (try
+      (when-let [ns-obj (find-ns (symbol ns-str))]
+        (let [c (compliment.context/cache-context context)
+              ^Class c (compliment.sources.class-members/try-get-object-class ns-obj c)]
+          (some-> c .getName)))
+      (catch Exception _
+        ;; We can always be analyzing a broken context.
+        nil))))
+
 (defn info
-  [{:keys [ns sym symbol class member] :as msg}]
-  (let [[ns sym class member] (map misc/as-sym [ns (or sym symbol) class member])
+  [{:keys [ns sym class member context]
+    legacy-sym :symbol
+    :as msg}]
+  (let [sym (or (not-empty legacy-sym)
```

Comment:
Why do you need `not-empty` here?

Reply:
For discarding empty strings

I don't necessarily expect them, but I've found it's a good habit to handle them.

Reply:
🤝

In an ideal world we'd have something like Spec checking at the edges